gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/ConditionSupplier.java

```diff
@@ -13,15 +13,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.gravitee.gateway.jupiter.core.condition;
-
-import io.gravitee.gateway.jupiter.api.context.RequestExecutionContext;
-import io.reactivex.Flowable;
+package io.gravitee.definition.model;
 
 /**
+ * Interface allowing to indicate that the entity can supply a condition.
+ *
  * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
  * @author GraviteeSource Team
  */
-public interface ConditionEvaluator<T> {
-    Flowable<T> filter(RequestExecutionContext ctx, Flowable<T> flows);
+@FunctionalInterface
+public interface ConditionSupplier {
+    /**
+     * Returns the condition or <code>null</code> if there is no condition defined.
+     * @return the condition or <code>null</code>.
+     */
+    String getCondition();
```

Comment:
why not proposing something to manage the absence of condition ?
```suggestion
    Optional<String> getCondition();
```

Reply:
It's a nice idea but it will break the existing classes that already define `getCondition()` as `String`. For compatibility it's preferable to keep using String and check for nullability.